Source/SwiftyDropbox/Platform/SwiftyDropbox_iOS/OAuthMobile.swift

```diff
@@ -343,10 +345,51 @@ open class MobileSharedApplication: SharedApplication {
     }
 
     open func presentAuthChannel(_ authURL: URL, tryIntercept: @escaping ((URL) -> Bool), cancelHandler: @escaping (() -> Void)) {
-        if let controller = self.controller {
-            let safariViewController = MobileSafariViewController(url: authURL, cancelHandler: cancelHandler)
-            controller.present(safariViewController, animated: true, completion: nil)
+        guard self.authChannel == nil else {
+            return
+        }
+        guard let controller = self.controller else {
+            return
+        }
+        
+        // ASWebAuthenticationSession and SFAuthenticationSession don't work with guided access (rdar://40809553)
```

Comment:
this comment is from AppAuth. I looked at the radar and it talks about security flaws, not sure how it's related to guided access. But figured might as well copy directly